### PR TITLE
Fix #3124: Activity Feed: Add support for deleting a post

### LINKED ISSUE #3124: Activity Feed: Add support for deleting a post
**Feature**
We should add support for deleting a post in both backend and frontend

- [ ] Backend
- [ ] Frontend - #3160

### PR DESCRIPTION
**Pull request by vivekratnavel**
### Describe your changes :

See #3124 

#
### Type of change :

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [ ] I have tagged my reviewers below.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/FeedRepository.java [hunk 1]
```diff
@@ -181,6 +187,38 @@ public Thread addPostToThread(String id, Post post) throws IOException {
...
+      if (userName.equals(post.getFrom())) {
+        List<Post> posts = thread.getPosts();
+        // Remove the post to be deleted from the posts list
+        posts = posts.stream().filter(p -> !p.getId().equals(UUID.fromString(postId))).collect(Collectors.toList());
+        thread.withUpdatedAt(System.currentTimeMillis()).withUpdatedBy(userName).withPosts(posts);
+        // update the json document
+        dao.feedDAO().update(threadId, JsonUtils.pojoToJson(thread));
+      } else {
+        // TODO Should we also allow admins to delete a post?
+        throw new AuthorizationException(CatalogExceptionMessage.noPermission(userName));
```
Comment (mithmatt) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/FeedRepository.java:214]: Minor (for future): We may need to surface better error message than the one from here..

"X cannot delete post by user Y" or something

Comment (harshach) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/FeedRepository.java:214]: Also allow admins to delete

Comment (harshach) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/FeedRepository.java:214]: any Auth checks should be delegated to the authorizer that way its easier enforce and not have exceptions

**Review by harshach — CHANGES_REQUESTED** [review]
Lets not merge the delete part before we make the permission checks to authorizer

**Comment by vivekratnavel** [comment]
@mithmatt @harshach Thank you for the reviews and feedback. I have delegated the authorization logic to the AuthZ. 

I will continue to work on converting the type of post.from and thread.createdBy to UUID. Then, the createPost will only require a message and we will populate post.from as id of the logged in user (securityContext.getUserPrincipal().getName()).

**Review by mithmatt — APPROVED** [review]
Thanks for addressing the review feedback  @vivekratnavel

File: catalog-rest-service/src/main/resources/json/schema/api/feed/createPost.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  "description": "Create post request",
+  "type": "object",
+  "properties": {
+    "message": {
+      "description": "Message in markdown format. See markdown support for more details.",
+      "type": "string"
+    },
+    "from": {
+      "description": "Name of the User posting the message",
+      "type": "string"
```
[the remark this answers is not in the document]
Comment (sureshms) [reply catalog-rest-service/src/main/resources/json/schema/api/feed/createPost.json:14]: Actually not. Let's keep them as user names which are supposed to be unique and human readable.

**Review by harshach — APPROVED** [review]

**Pull request merged**